server/datastore/mysql/policies_test.go

```diff
@@ -2277,3 +2280,105 @@ func testUpdatePolicyFailureCountsForHosts(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 	require.Len(t, hostsUpdated, 0)
 }
+
+func testListGlobalPoliciesCanPaginate(t *testing.T, ds *Datastore) {
+	// create 30 policies
+	for i := 0; i < 30; i++ {
+		_, err := ds.NewGlobalPolicy(context.Background(), nil, fleet.PolicyPayload{Name: fmt.Sprintf("policy%d", i)})
+		require.NoError(t, err)
+	}
+
+	// Page 0 contains 20 policies
+	policies, err := ds.ListGlobalPolicies(context.Background(), fleet.ListOptions{
+		Page:    0,
+		PerPage: 20,
+	})
+
+	assert.Equal(t, "policy0", policies[0].Name)
+	assert.Len(t, policies, 20)
+	require.NoError(t, err)
+
+	// Page 1 contains 10 policies
+	policies, err = ds.ListGlobalPolicies(context.Background(), fleet.ListOptions{
+		Page:    1,
+		PerPage: 20,
+	})
+
+	assert.Equal(t, "policy20", policies[0].Name)
+	assert.Len(t, policies, 10)
+	require.NoError(t, err)
+
+	// No list options returns all policies
+	policies, err = ds.ListGlobalPolicies(context.Background(), fleet.ListOptions{})
+	assert.Len(t, policies, 30)
+	require.NoError(t, err)
+}
+
+func testListTeamPoliciesCanPaginate(t *testing.T, ds *Datastore) {
+	tm, err := ds.NewTeam(context.Background(), &fleet.Team{Name: "team1"})
+	require.NoError(t, err)
+
+	// create 30 policies
+	for i := 0; i < 30; i++ {
+		_, err := ds.NewTeamPolicy(context.Background(), tm.ID, nil, fleet.PolicyPayload{Name: fmt.Sprintf("policy%d", i)})
+		require.NoError(t, err)
+	}
```

Comment:
Just in case, let's also create dummy global policies too (to know that the querying/filtering is working correctly)